cocos/particle/renderer/particle-system-renderer-data.ts

```diff
@@ -330,12 +323,7 @@ export default class ParticleSystemRenderer {
             this._particleSystem.processor = null!;
         }
         const useGPU = this._useGPU && isSupportGPUParticle();
-        if (!useGPU && this.cpuMaterial) {
-            this.particleMaterial = this.cpuMaterial;
-        }
-        if (useGPU && this.gpuMaterial) {
-            this.particleMaterial = this.gpuMaterial;
-        }
+        this.particleMaterial = useGPU ? this.gpuMaterial : this.cpuMaterial;
```

Comment:
Should check whether `this.particleMaterial` is null?

Reply:
No need, after setting, subsequent logic includes checks. 

Related API: _onMaterialModified